### PR TITLE
backend for the example app written in python

### PR DESCRIPTION
**Pull request by chaudum**
includes a universal integration test written in python

### CODE REVIEW

File: python/README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+Create a virtualenv:
+
+```console
+$ virtualenv env
+$ source ./venv/bin/activate
+$ ./env/bin/pip install -r requirements.txt
+```
+
+Create tables and populate them with test data:
+
```
Comment (chaudum) [inline python/README.md]: reference README in root folder

File: python/app.py [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+from flask import (
+    Flask,
+    g as app_globals,
+    make_response,
+)
+from flask.ext.restful import Api, Resource, reqparse
+
+
+# app configuration
+MAX_CONTENT_LENGTH = 1024 * 1024  # 1mb
```
Comment (chaudum) [inline python/app.py]: not used

File: python/app.py [hunk 3]
```diff
@@ -0,0 +1,366 @@
...
+
+class Post(CrateResource, PostResource):
+    """
+    Resource for guestbook.posts
+    Supported methods: GET, PUT, DELETE
+    """
+
+    def __init__(self):
+        super(Post, self).__init__()
+        self.reqparse = reqparse.RequestParser()
```
Comment (chaudum) [inline python/app.py]: only used in `PUT`

File: python/app.py [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+        # json-serializable object array
+        response = self.convert(self.cursor.description,
+                                self.cursor.fetchall())
+        return (response, 200)
+
+    def post(self):
+        # parse incoming POST data
+        args = self.reqparse.parse_args()
+        if not args.get('text'):
+            return self.error('Argument "text" is required!', 400)
```
Comment (chaudum) [inline python/app.py]: make separate method for "argument required"

File: python/app.py [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+
+    __name__ = 'Image'
+    __table__ = 'blob.guestbook_images'
+    CONTENT_TYPE = 'image/gif'
+
+    def _blob_url(self, digest):
+        return '{}://{}/_blobs/{}/{}'.format('http', CRATE_HOST,
+                                             self.__table__.split('.')[1],
+                                             digest)
+
```
Comment (chaudum) [inline python/app.py]: one newline too much

File: python/app.py [hunk 6]
```diff
@@ -0,0 +1,366 @@
...
+class ImageList(CrateResource, ImageResource):
+    """
+    Resource for blob.guestbook_images
+    Supported methods: POST, GET
+    """
+
+    def __init__(self):
+        super(ImageList, self).__init__()
+        self.reqparse = reqparse.RequestParser()
```
Comment (chaudum) [inline python/app.py]: only used in `POST` - move it there

File: sql/populate.sql [hunk 7]
```diff
@@ -0,0 +1,7 @@
+INSERT INTO guestbook.countries (name, coordinates)
```
Comment (chaudum) [inline sql/populate.sql]: remove file

File: sql/schemas.sql [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+-- countries table
+DROP TABLE IF EXISTS guestbook.countries;
+CREATE TABLE IF NOT EXISTS guestbook.countries (
+    id STRING PRIMARY KEY,
+    name STRING PRIMARY KEY,
+    geometry GEO_SHAPE
+) WITH (number_of_replicas='0-2');
+
+-- images blob table
+DROP BLOB TABLE IF EXISTS blob.guestbook_images;
```
Comment (chaudum) [inline sql/schemas.sql]: `IF EXISTS` does not work with blob tables

File: python/app.py [hunk 9]
```diff
@@ -0,0 +1,366 @@
...
+    Flask,
+    g as app_globals,
+    make_response,
+)
+from flask.ext.restful import Api, Resource, reqparse
+
+
+# app configuration
+MAX_CONTENT_LENGTH = 1024 * 1024  # 1mb
+CRATE_HOST = 'localhost:4200'
```
Comment (joemoe) [inline python/app.py]: we are having a config file in php. don't you also want to put that stuff there?

**Comment by joemoe** [comment]
ack

**Comment by kovrus** [comment]
ack

**Pull request merged**